engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/EmbeddedDeploymentManager.scala

```diff
@@ -137,7 +137,7 @@ class EmbeddedDeploymentManager(modelData: ModelData, engineConfig: Config,
       case Some(ScenarioInterpretationData(_, _, interpreterTry)) => Future.successful {
         interpreters -= name
         interpreterTry.foreach { interpreter =>
-          interpreter.close()
+          deploymentStrategy.onScenarioCancelled(interpreter)
```

Comment:
Won't be easier to have "something" smarter as a value in map of interpreters which would be created in  `onScenarioAdded` instead of having such many methods in  `DeploymentStrategy`? It looks like ping-pong now. U invoke strategy to return some object and then you pass this object again to the strategy.

Reply:
:+1: good idea, refactored a bit